docs/guide.md

```diff
@@ -142,6 +142,16 @@ API keys can be created, viewed and revoked programmatically by manipulating the
 !!! danger
     To prevent leaking API keys, you must only give the `key` **to the client that triggered its generation**. In particular, **do not keep any trace of it on the server**.
 
+- To retrieve an `APIKey` instance based on its generated key (which is not stored in the database) use the `.get_from_key()` method on the `APIKey` objects manager instead of `.get()`:
```

Comment:
Should we perhaps add a comment about what use cases this addresses? Eg, if this is accurate…

```suggestion
- To retrieve an `APIKey` instance based on its generated key (which is not stored in the database) use the `.get_from_key()` method on the `APIKey` objects manager instead of `.get()`. This is useful if you'd like to access an `APIkey` object from a view protected by a `HasAPIKey` permission.
```

And then maybe update the code snippet to demonstrate that use case?

```python
class ProjectListView(APIView):
    permission_classes = [HasAPIKey]

    def get(self, request):
        """Retrieve a project based on the request API key."""
        key = request.META["HTTP_AUTHORIZATION"]
        api_key = APIKey.get_from_key(key)
        project = Project.objects.get(api_key=api_key)
```

This roughly corresponds to the example in https://github.com/florimondmanca/djangorestframework-api-key/issues/20#issuecomment-487454993 (which I'm super glad we finally have a proper built-in API for!).

(Note that the usage of `request.META` is duplicated from `HasAPIKey`, and it might be wrong in some cases, e.g. if we customized the API key header. Ideally we should use `HasAPIKey.get_key(request)`, but that's an instance method for now. Probably worth considering converting that to an `@classmethod`…)

Reply:
I like this code snippet a lot, much clearer! I added your note about typical use-cases and updated the code snippet below to match your suggested snippet. I didn't leave any notes about adjusting the `request.META` access in case of a customized key header, but let me know if that would be helpful and I can add a comment to the snippet.

Reply:
The docs try to stay pretty much agnostic of customization outside of the "Customization" section, so it's probably okay to assume we're in the default setup and the API key is in the `Authorization` header. So I think we're good!